File: algorithms/algorithms-cplusplus/leetcode/Question_0977_Squares_of_a_Sorted_Array.cc
#include <vector>

using namespace std;

class Solution {
 public:
  vector<int> sortedSquares(vector<int>& nums) {
    vector<int> res(nums.size(), 0);
    int l = 0, r = nums.size() - 1;
    for (int i = res.size() - 1; i >= 0; i--) {
      if (abs(nums[r]) >= abs(nums[l])) {
        res[i] = nums[r] * nums[r];
        r--;
      } else {
        res[i] = nums[l] * nums[l];
        l++;
      }
    }
    return res;
  }
};

File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0977_Squares_of_a_Sorted_Array.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0977_Squares_of_a_Sorted_Array {
    public int[] sortedSquares(int[] nums) {
        int[] res = new int[nums.length];
        int l = 0, r = nums.length - 1;
        for (int i = res.length - 1; i >= 0; i--) {
            if (Math.abs(nums[r]) >= Math.abs(nums[l])) {
                res[i] = nums[r] * nums[r];
                r--;
            } else {
                res[i] = nums[l] * nums[l];
                l++;
            }
        }
        return res;
    }
}
